neqo-http3/src/client_events.rs

```diff
@@ -240,11 +240,11 @@ impl ExtendedConnectEvents for Http3ClientEvents {
         Ok(())
     }
 
-    fn new_datagram(&self, session_id: StreamId, datagram: Vec<u8>) {
+    fn new_datagram(&self, session_id: StreamId, datagram: VecDeque<u8>) {
         self.insert(Http3ClientEvent::WebTransport(
             WebTransportEvent::Datagram {
                 session_id,
-                datagram,
+                datagram: datagram.into(),
```

Comment:
This would re-allocate the `VecDeque` as a `Vec`, correct? Thus the performance hit just moves into `client_events.rs`?

I guess we could emit a `VecDeque` in the event itself, i.e. `WebTransportEvent::Datagram`'s `datagram`.